### PR TITLE
Fix stress corruption caused by water #552

### PR DESCRIPTION
**Pull request by curiosity-a**

### CODE REVIEW

File: map_gen/maps/diggy/template.lua [hunk 1]
```diff
@@ -41,28 +42,48 @@ local function insert_next_tiles(data)
...
                 end
             end
+            if string_find(new_tile.name, "water", 1, true) then --maybe check prototype's collision mask instead?
+                local entities = surface.find_entities{{tile_pos.x, tile_pos.y}, {tile_pos.x + 1, tile_pos.y + 1}}
+                for _, entity in pairs(entities) do
+                    if entity.teleport({entity.position.x + teleport_offset, entity.position.y}) then
+                        teleported_count = teleported_count + 1
+                        teleported[teleported_count] = entity
+                    else
+                        print("WARNING: entity " .. entity.name .. " failed to teleport out of the way at " .. serpent.line(entity.position))
```
Comment (grilledham) [inline map_gen/maps/diggy/template.lua]: What does `print` refer to? If it's `G.print` then you probably didn't want that. Maybe use `log` instead.

File: map_gen/maps/diggy/template.lua [hunk 2]
```diff
@@ -41,28 +42,48 @@ local function insert_next_tiles(data)
...
             if new_tile.name ~= 'out-of-map' then
-                local current_tile = get_tile(new_tile.position.x, new_tile.position.y)
+                local current_tile = get_tile(tile_pos.x, tile_pos.y)
                 if current_tile.name == 'out-of-map' then
                     void_removed_count = void_removed_count + 1
                     void_removed[void_removed_count] = {surface = surface, position = current_tile.position}
                 end
             end
+            if string_find(new_tile.name, "water", 1, true) then --maybe check prototype's collision mask instead?
+                local entities = surface.find_entities{{tile_pos.x, tile_pos.y}, {tile_pos.x + 1, tile_pos.y + 1}}
```
Comment (grilledham) [inline map_gen/maps/diggy/template.lua]: To be consistent with the other surface calls, we could make a local `find_entities`

File: map_gen/maps/diggy/template.lua [hunk 3]
```diff
@@ -41,28 +42,48 @@ local function insert_next_tiles(data)
...
+                for _, entity in pairs(entities) do
+                    if entity.teleport({entity.position.x + teleport_offset, entity.position.y}) then
```
Comment (grilledham) [inline map_gen/maps/diggy/template.lua]: I think you can just call this as `entity.teleport(teleport_offset)`
Then do the same thing below.

Saves making a table, but arguably your way is clearer.

File: map_gen/maps/diggy/template.lua [hunk 4]
```diff
@@ -41,28 +42,48 @@ local function insert_next_tiles(data)
...
-                local current_tile = get_tile(new_tile.position.x, new_tile.position.y)
+                local current_tile = get_tile(tile_pos.x, tile_pos.y)
                 if current_tile.name == 'out-of-map' then
                     void_removed_count = void_removed_count + 1
                     void_removed[void_removed_count] = {surface = surface, position = current_tile.position}
                 end
             end
+            if string_find(new_tile.name, "water", 1, true) then --maybe check prototype's collision mask instead?
+                local entities = surface.find_entities{{tile_pos.x, tile_pos.y}, {tile_pos.x + 1, tile_pos.y + 1}}
+                for _, entity in pairs(entities) do
```
Comment (grilledham) [inline map_gen/maps/diggy/template.lua]: For new code we seem to have adopted the style of using `for i = 1, #entities` over using `pairs`.

**Review by grilledham — APPROVED** [review]
I think this is a decent solution to the bug. 
Someone who is more experienced with the diggy code might be able to give a more meaningful review though.

**Review by linaori — APPROVED** [review]

**Comment by grilledham** [comment]
I'm merging now, thanks for the PR.
Closes 552

**Pull request merged**